pymatgen/core/surface.py

```diff
@@ -214,6 +214,44 @@ def surface_area(self) -> float:
         matrix = self.lattice.matrix
         return np.linalg.norm(np.cross(matrix[0], matrix[1]))
 
+    def is_symmetric(self, symprec: float = 0.1) -> bool:
```

Comment:
`is_symmetric` and `is_polar` look very much like a `property`. Wondering if there is a way to mark it as `property` and take an argument at the same time. (maybe carry a global `tolerance` attrib? As many downstream operations like `get_tasker2_slabs`, `get_slab`/`get_slabs` from `SlabGenerator` all need a tolerance)

Reply:
i'm usually not in favor of extra state like `SlabGenerator.tolerance` unless really necessary. state tends to make code more confusing

i don't think you can make these into `@property`s without a breaking change either. perhaps best to leave as is?

Reply:
> i'm usually not in favor of extra state like `SlabGenerator.tolerance` unless really necessary. state tends to make code more confusing

Okay let's keep it as is. There're more urgent issues to address for `surface` than this.

> i don't think you can make these into `@property`s without a breaking change either. perhaps best to leave as is?

Yes sure. I would avoid breaking changes unless necessary.